### PR TITLE
client: add firmware extensions|interfaces command (jsc#PED-3132)

### PR DESCRIPTION
**Pull request by mtomaschewski**
- `wicked firmware extensions` shows available firmware extensions table:
  ```
  ibft          enabled
  redfish       enabled
  ```
- `wicked firmware interfaces` shows interfaces configured by a firmware:
  ```
  ibft          eth1 eth1.10
  redfish       usb0
  ```

See also https://github.com/openSUSE/installation-images/pull/620

### CODE REVIEW

File: client/firmware.c [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	argv0 = argv[0];
+	argv[0] = command;
+	optind = 1;
+	while ((opt = getopt_long(argc, argv, "+h", options, NULL))  != -1) {
+		switch (opt) {
+		case OPTION_HELP:
+			status = NI_WICKED_RC_SUCCESS;
+			/* fall through */
+		default:
+		usage:
```
Comment (cfconrad) [inline client/firmware.c:67]: I think we should add the goto jumppoint to the start of the line.
```suggestion
usage:
```

Comment (mtomaschewski) [reply client/firmware.c:67]: This is how we usually do this, see e.g. wicked client main():
```
                switch (c) {
                case OPT_HELP:
                        status = NI_WICKED_ST_OK;
                        /* fall through */
                default:
                usage:
```

Comment (cfconrad) [reply client/firmware.c:67]: ok

**Review by cfconrad — APPROVED** [review]
LGTM

**Pull request merged**